src/uct/ib/dc/dc_mlx5.c

```diff
@@ -199,12 +200,15 @@ static ucs_status_t uct_dc_mlx5_iface_query(uct_iface_h tl_iface, uct_iface_attr
                                    max_am_inline,
                                    UCT_RC_MLX5_TM_EAGER_ZCOPY_MAX_IOV(UCT_IB_MLX5_AV_FULL_SIZE));
 
-    if (uct_ib_mlx5_iface_has_ar(&iface->super.super.super) &&
-        (!(iface->flags & (UCT_DC_MLX5_IFACE_FLAG_DCI_FULL_HANDSHAKE |
-                           UCT_DC_MLX5_IFACE_FLAG_KEEPALIVE_FULL_HANDSHAKE |
-                           UCT_DC_MLX5_IFACE_FLAG_DCT_FULL_HANDSHAKE)))) {
-        iface_attr->cap.flags &= (~UCT_IFACE_FLAG_PUT_BCOPY & ~UCT_IFACE_FLAG_PUT_ZCOPY);
+    is_full_handshake = iface->flags &
+                        (UCT_DC_MLX5_IFACE_FLAG_DCI_FULL_HANDSHAKE |
+                         UCT_DC_MLX5_IFACE_FLAG_KEEPALIVE_FULL_HANDSHAKE |
+                         UCT_DC_MLX5_IFACE_FLAG_DCT_FULL_HANDSHAKE);
```

Comment:
should be 
ucs_test_all_flags(iface->flags, UCT_DC_MLX5_IFACE_FLAG_DCI_FULL_HANDSHAKE|UCT_DC_MLX5_IFACE_FLAG_DCT_FULL_HANDSHAKE)

because:
1. we don't care about keepalive DCI, we don't post PUT ops on it
2. need both DCI and DCT full handshake (or just DCI) to enable PUT